app/main/routes.py

```diff
@@ -590,132 +543,66 @@ def search_results_summary():
         if search_filter:
             search_terms.append(search_filter)
-            query += (
-                "," + search_filter if len(search_terms) > 0 else search_filter
-            )
-            filters = {"query": query}
-
-        breadcrumb_values.update({0: {"query": query}})
+            query += f",{search_filter}" if search_terms else search_filter
+            filters["query"] = query
 
-        display_terms = ""
-
-        for i in range(len(search_terms)):
-            if len(search_terms[i].strip()) > 0:
-                display_terms += (
-                    "‘"
-                    + search_terms[i].strip()
-                    + "’"
-                    + (" + " if i < len(search_terms) - 1 else "")
-                )
-
-        breadcrumb_values.update(
-            {3: {"search_terms": query if query == "," else display_terms}}
+        breadcrumb_values[0] = {"query": query}
+        display_terms = " + ".join(
+            [f"‘{term}’" for term in search_terms if term.strip()]
         )
+        breadcrumb_values[3]["search_terms"] = display_terms or query
 
-        open_search = OpenSearch(
-            hosts=current_app.config.get("OPEN_SEARCH_HOST"),
-            http_auth=current_app.config.get("OPEN_SEARCH_HTTP_AUTH"),
-            use_ssl=True,
-            verify_certs=True,
-            ca_certs=current_app.config.get("OPEN_SEARCH_CA_CERTS"),
-            connection_class=RequestsHttpConnection,
-        )
-
-        dsl_query = {
-            "query": {
-                "bool": {
-                    "must": [
-                        {
-                            "multi_match": {
-                                "fields": ["*"],
-                                "query": query,
-                                "fuzziness": 1,
-                            }
-                        }
-                    ],
-                    "filter": [{"term": {"transferring_body_id.keyword": _id}}],
-                },
-            },
-            "highlight": {
-                "pre_tags": ["<mark>"],
-                "post_tags": ["</mark>"],
-                "fields": {
-                    "*": {},
-                },
-            },
-            "sort": sorting_query,
-        }
-        size = per_page
-        page_number = page
-        from_ = size * (page_number - 1)
-        try:
-            search_results = open_search.search(
-                dsl_query,
-                from_=from_,
-                size=size,
-                timeout=current_app.config["OPEN_SEARCH_TIMEOUT"],
-            )
-        except opensearchpy.exceptions.ConnectionTimeout:
-            abort(504)
+        open_search = setup_opensearch()
+        dsl_query = build_dsl_query(query, search_area, open_search, _id)
 
+        search_results = execute_search(open_search, dsl_query, page, per_page)
         results = search_results["hits"]["hits"]
 
         for result in results:
```

Comment:
can we extract this for loop into a format_opensearch_results function as we're doing a refactor?